python/ray/serve/api.py

```diff
@@ -638,7 +642,70 @@ async def __call__(self, request):
         number if you want to save memory on the node resource.
     """
 
-    raise NotImplementedError("Multiplexed deployment is not supported yet.")
+    if func is not None:
+        if not callable(func):
+            raise TypeError(
+                "The `multiplexed` decorator must be used with a function or method."
+            )
+
+        if not inspect.iscoroutinefunction(func):
+            raise TypeError(
+                "@serve.multiplexed can only be used to decorate async "
+                "functions or methods."
+            )
```

Comment:
we should remove this requirement and post sync callables to an executor (can be a follow-up PR)

Reply:
would like to do the same for batch handler

Reply:
Leave it to as a  followup :)

Reply:
ok please leave TODO + link to a github issue tracking it